lib/Parse/ParseType.cpp

```diff
@@ -230,6 +238,12 @@ ParserResult<TypeRepr> Parser::parseTypeSimple(
     }
     return makeParserCodeCompletionResult<TypeRepr>(
         ErrorTypeRepr::create(Context, consumeToken(tok::code_complete)));
+  case tok::integer_literal: {
+    auto text = copyAndStripUnderscores(Tok.getText());
+    auto loc = consumeToken(tok::integer_literal);
+    ty = makeParserResult(new (Context) IntegerTypeRepr(text, loc, minusLoc));
+    break;
+  }
```

Comment:
What kind of error message do we generate for the following code now?

```swift
let x: 1
```

or

```swift
let x: -1
```

Reply:
These are now diagnosed during type checking:
```
/Users/alejandro.alonso/6.swift:1:8: error: integer unexpectedly used in a type position
1 | let x: 1
  |        `- error: integer unexpectedly used in a type position
2 | let y: -1

/Users/alejandro.alonso/6.swift:2:9: error: integer unexpectedly used in a type position
1 | let x: 1
2 | let y: -1
  |         `- error: integer unexpectedly used in a type position
```